pkg/istio/authorizationpolicy/controller.go

```diff
@@ -0,0 +1,360 @@
+	}
+
+	athenzDomain, ok := athenzDomainRaw.(*adv1.AthenzDomain)
+	if !ok {
+		return errors.New("athenz domain cast failed, domain name: " + athenzDomainName)
+	}
+
+	signedDomain := athenzDomain.Spec.SignedDomain
+	domainRBAC := athenz.ConvertAthenzPoliciesIntoRbacModel(signedDomain.Domain, &c.adIndexInformer)
+	var serviceList []*corev1.Service
+	if serviceName != "" {
+		serviceObj, err := c.getSvcObj(athenz.DomainToNamespace(athenzDomainName) + "/" + serviceName)
+		if err != nil {
+			return fmt.Errorf("error getting service from cache: %s", err.Error())
+		}
+		if serviceObj != nil {
+			serviceList = append(serviceList, serviceObj)
+		}
+	} else {
+		// handle case 2
+		// fetch all services in the namespace
+		// error checking
+		// add to serviceList
+		svcKeys := c.serviceIndexInformer.GetStore().ListKeys()
+		domainName := athenz.DomainToNamespace(athenzDomainName)
+		for _, svcKey := range svcKeys {
+			if strings.Split(svcKey, "/")[0] == domainName {
+				serviceObj, err := c.getSvcObj(svcKey)
+				if err != nil {
+					return fmt.Errorf("error getting service from cache: %s", err.Error())
+				}
+				if serviceObj != nil {
+					serviceList = append(serviceList, serviceObj)
+				}
+			}
+		}
+	}
+
+	var desiredCRs []model.Config
+	// range over serviceList
+	for _, service := range serviceList {
+		// if svc annotation authz.istio.io/enabled is not present - skip processing and continue
+		if !c.checkAuthzEnabledAnnotation(service) {
+			continue
+		}
+		desiredCR := c.rbacProvider.ConvertAthenzModelIntoIstioRbac(domainRBAC, service.Name, service.Labels["svc"])
+		// append to desiredCRs array
+		desiredCRs = append(desiredCRs, desiredCR...)
+	}
+
+	// get current APs from cache
+	currentCRs := c.rbacProvider.GetCurrentIstioRbac(domainRBAC, c.configStoreCache, serviceName, c.dryRun)
+	cbHandler := c.getCallbackHandler(key)
+	changeList := common.ComputeChangeList(currentCRs, desiredCRs, cbHandler, c.checkOverrideAnnotation)
+
+	// If change list is empty, nothing to do
+	if len(changeList) == 0 {
+		log.Infof("Everything is up-to-date for key: %s", key)
+		c.queue.Forget(key)
+		return nil
+	}
+
+	for _, item := range changeList {
+		log.Infof("Adding resource action to queue: %s on %s for key: %s", item.Operation, item.Resource.Key(), key)
+		c.processConfigChange(item)
+	}
+	return nil
+}
+
+// checkOverrideAnnotation checks if current config has override annotation, skips process if override annotation is set
+// to true
+func (c *Controller) checkOverrideAnnotation(existingConfig model.Config) bool {
+	if res, exists := existingConfig.ConfigMeta.Annotations[overrideAnnotation]; exists {
+		return res == "true"
+	} else {
+		return exists
+	}
+}
+
+// ProcessConfigChange receives resource and event action, and perform update on resource
+func (c *Controller) processConfigChange(item *common.Item) error {
+	if item == nil {
+		return nil
+	}
+	var err error
+	var eHandler common.EventHandler
```

Comment:
Ah nevermind, I see that this will need to change based on if the dry run is enabled per service. Does it make sense to create both the dry run and api handler objects in the controller then and then call the corresponding one based on if it's in dry run mode or not for that particular service?